contribs/gmf/src/directives/search.js

```diff
@@ -0,0 +1,272 @@
+/**
+ * @fileoverview this file provides the "search" directive and controller for
+ * a GeoMapFish application.
+ *
+ * FIXME: The 'placeholder' in the input field is hard-coded and can't currently
+ * be internationalised.
+ *
+ * Example:
+ *
+ * <gmf-search gmf-search-map="ctrl.map"
+ *             gmf-search-datasources="ctrl.searchDatasources">
+ * </gmf-search>
+ */
+goog.provide('gmf.searchDirective');
+
+
+goog.require('gmf');
+goog.require('gmf.mapDirective');
+goog.require('ngeo.CreateGeoJSONBloodhound');
+goog.require('ngeo.FeatureOverlay');
+goog.require('ngeo.FeatureOverlayMgr');
+goog.require('ngeo.searchDirective');
+goog.require('ol.Map');
+goog.require('ol.View');
+goog.require('ol.layer.Tile');
+goog.require('ol.proj');
+
+
+/**
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ */
+gmf.searchDirective = function() {
+  return {
+    restrict: 'E',
+    scope: {
+      'getMapFn': '&gmfSearchMap',
+      'getDatasourcesFn': '&gmfSearchDatasources'
+    },
+    controller: 'GmfSearchController',
+    controllerAs: 'ctrl',
+    template:
+        '<input type="text" placeholder="search…" ' +
+        'ngeo-search="ctrl.options" ' +
+        'ngeo-search-datasets="ctrl.datasets" ' +
+        'ngeo-search-listeners="ctrl.listeners">',
+    link:
+        /**
+         * @param {angular.Scope} scope Scope.
+         * @param {angular.JQLite} element Element.
+         * @param {angular.Attributes} attrs Atttributes.
+         */
+        function(scope, element, attrs) {
+          // Empty the search field on focus and blur.
+          element.find('input').on('focus blur', function() {
+            $(this).val('');
+          });
+        }
+  };
+};
+
+gmfModule.directive('gmfSearch', gmf.searchDirective);
+
+
+
+/**
+ * @constructor
+ * @param {angular.Scope} $scope The directive's scope.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {ngeo.CreateGeoJSONBloodhound} ngeoCreateGeoJSONBloodhound The ngeo
+ *     create GeoJSON Bloodhound service.
+ * @param {ngeo.FeatureOverlayMgr} ngeoFeatureOverlayMgr The ngeo feature
+ *     overlay manager.
+ * @ngInject
+ * @export
+ */
+gmf.SearchController = function($scope, $compile,
+    ngeoCreateGeoJSONBloodhound, ngeoFeatureOverlayMgr) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {angular.$compile}
+   * @private
+   */
+  this.compile_ = $compile;
+
+  /**
+   * @type {ngeo.CreateGeoJSONBloodhound}
+   * @private
+   */
+  this.ngeoCreateGeoJSONBloodhound_ = ngeoCreateGeoJSONBloodhound;
+
+  var map = this.scope_['getMapFn']();
+  goog.asserts.assertInstanceof(map, ol.Map);
+
+  /**
+   * @type {!ol.Map}
+   * @private
+   */
+  this.map_ = map;
+
+  ngeoFeatureOverlayMgr.init(this.map_);
```

Comment:
This is not correct. This should be done at the application level, in the main controller.